samples/python/reactors/PorousMediaBurner.py

```diff
@@ -0,0 +1,395 @@
+Given a reaction mechanism, equations for species mass
+fractions, gas phase temperature and solid temperature are solved.
+The reactor cascade consists of a variable number of cylindrical reactors.
+The Rosseland model is used for radiative heat transfer by the solid.
+A Nusselt number model is used to compute the heat transfer coefficient.
+This example assumes finding steady-state solutions with constant pressure
+so that mass controllers and pressure valves are not required. The code is written in
+a general way to support an arbitrary number of reactors and an arbirary number of
+burner sections with different physical properties.
+
+The implemented equations make use of Cantera's extensible reactor models.
+Therefore, Cantera version 2.6.0a4 or higher is required.
+
+The porous media burner considered in this example is a cylindrical tube filled
+three different porous materials: a porous ceramic made from a Yttria-stabilized
+Zirconia Alumina (YZA) section with length of 2 inches and pore density of 40 pores per
+inches (PPI), then a one inch section with a porous ceramic made from
+silicon carbide (SiC) with 3 PPI and finally a 1 inch section of SiC with 10 PPI. A
+flame stabilizes at the interface between the YZA and 3 PPI SiC.
+
+
+          |--- two inches --- | --- one inch ---| --- one inch --- |
+          __________________________________________________________
+fuel
+ +   =>      YZA 40 PPI       |     SiC 3 PPI   |   SiC 10 PPI       => burnt
+air       __________________________________________________________    gas
+
+                inert            flame location   heat recirculation
+
+
+More details about the governing equations, submodels and physical setup and
+corresponding experiments can be found in
+
+    Experimental and numerical investigation of flame stabilization and
+    pollutant formation in matrix stabilized ammonia-hydrogen combustion,
+    G. Vignat, T. Zirwes, E.R. Toro, K. Younes, E. Boigne, P. Muhunthan,
+    L. Simitz, D. Trimis, M. Ihme
+    Combustion and Flame (https://doi.org/10.1016/j.combustflame.2023.112642)
+
+Requires: cantera >= 2.6.0a4
+Keywords: porous media burner, heat transfer between gas and solid,
+          radiation modeling, reactor cascade
+"""
+
+import matplotlib.pyplot as plt
+import cantera as ct
+import numpy as np
+
+inch = 0.0254          # conversion of inches to meters
+Tamb = 300.            # ambient temperature (used for radiation modeling) (K)
+TsInit = 300.          # initial temperature of the solid (K)
+constantP = ct.one_atm # gas pressure inside the porous medium (Pa)
+mech = "gri30.yaml"    # reaction mechanism file
+phi = 1.2              # equivalence ratio (phi) of the fuel/oxidizer mixture
+mdot = 0.3             # mass flux through the cascade (assumed constant) (kg/m^2/s)
+                       # with respect to the cross-sectional area of the burner
+
+# initial state of the gas phase containing fuel and oxidizer at the inlet
+def getInitalGas(gas):
```

Comment:
```suggestion
def getInitialGas(gas):
```
And of course all places where this is called.